pytorch/src/main/java/org/bytedeco/pytorch/presets/torch.java

```diff
@@ -311,22 +320,26 @@ public void map(InfoMap infoMap) {
             .put(new Info().javaText("import org.bytedeco.pytorch.Allocator;"))
             .put(new Info().javaText("import org.bytedeco.pytorch.Function;"))
             .put(new Info().javaText("import org.bytedeco.pytorch.functions.*;"))
+            .put(new Info().javaText("import org.bytedeco.pytorch.chrono.*;"))
             .put(new Info().javaText("import org.bytedeco.pytorch.Module;"))
             .put(new Info().javaText("import org.bytedeco.javacpp.annotation.Cast;"))
+            .put(new Info().javaText("import org.bytedeco.pytorch.presets.torch.IntrusivePtr;"))
```

Comment:
Please put those in the helper package and make the presets depend on that

Reply:
Done

Reply:
Actually, we only need packages when there is a name conflict. Why do you want to put things in different packages??

Reply:
Which package are you talking about ? 
New packages in this PR are:
*  gloo: this is a "third-party" independent piece of software bundled into pytorch. It wont't be used by 99% of users. It contains simple class names like "Device", "Buffer"...  that would conflict with Pytorch names unless renamed.
* chrono : these are manual classes for binding some std::chrono classes. Again this is separate software from Pytorch. It won't be used by most people either. It could eventually be moved to another presets if completed enough.

Main package is already quite crowded, and I'm sure there are some names shadowed by others. If redesigning from scratch I would  suggest to move some of torch namespaces that are well separated and targeted for some specific uses to separate Java packages, like `torch::jit`.

Reply:
( I was tempted to use a separate package for `distributed` (`c10d` namespace) but I knew you won't like that :). Also it would have really needed 2 packages: `distributed` and `cuda.distributed`. So I prefered to put them in main packages.)

Reply:
Like all of them, if you want to make packages, at least make them related to the C++ API and not just come up with random names like you're doing. Like where does "cuda" and "functions" come from? What do they correspond to exactly with the C++ API?

Reply:
`cuda` contains the classes than must only by used if you have the `-gpu` classifier. I think it's a good thing to have them in a separate package. It contains classes from the `c10::cuda` namespace and some dependency classes.
`function` contains Java mapping for the `std::function` instances used by the presets. They don't map to Pytorch C++ names.
Where is randomness here ?
But if you want me to merge the function package into the main one, I won't mind.

Reply:
The convention that I've been trying to keep for that is to match the library names. We already have "torch" and "torch_cuda" as libraries, so we should be using that. However, I understand that you wanted to not break backward compatibility so you came up with torch.cuda, so that's find I guess because torch::cuda exists, but there's no torch::functions or torch::gloo. Like I keep telling you, if you want to create some other API, go do it somewhere else. The idea here is to have the closest possible mapping to the C++ API. Please do that

Reply:
About chrono: what about creating an independent presets for that ?
I'm seeing that depthai and tesseract use some chrono classes and the presets map them to `Pointer`.

About gloo: if I move them to main packages, I'll have to prefix all names by `Gloo`. Is that really better ? I remind you that this is not pytorch code. I can first only keep the classes that are strictly necessary when used from PyTorch.

About function: ok, I'll remove these packages. This could break some code of users of last versions, they will have to remove their import.

Reply:
The point is about making this consistent with the C++ API, so like we could put std:: function in "function", std::chrono in "chrono", torch::jit in "jit", whatever::gloo in "gloo", that's fine. Would you be ok following this pattern?

Reply:
The namespace structure of PyTorch is quite messy. We have:
```
at
at::cuda
at::functorch
at::impl
at::indexing
at::internal
at::namedinference
at::native
c10
c10::cuda
c10d
c10::detail
c10::detail::infer_schema
c10d::intra_node_comm
c10::impl
c10::ivalue
c10::util
c10::utils
caffe2
caffe2::serialize
torch
torch::autograd
torch::data
torch::data::datasets
torch::data::example
torch::data::samplers
torch::dynamo::autograd
torch::enumtype
torch::inductor
torch::jit
torch::jit::detail
torch::jit::interpreter
torch::nn
torch::nn::detail
torch::nn::functional
torch::nn::utils::rnn
torch::optim
torch::profiler::impl
torch::serialize
torch::utils
```
Some of them could be merged since PyTorch defined namespace aliases if I remember well, but it doesn't seem feasible without support in JavaCPP to automatically map namespaces to Java packages.

What we can do today is to target a Java package when we parse a set of header files. So, rather than namespaces, Java packages must match either independent/thirdparty API used by the library (like gloo or std:chrono), or subsets of the library API that is only enabled when including some specific headers (like cuda, maybe jit, maybe distributed/c10d).

Reply:
Yes, that's why I keep saying let's not do packages. You're the one that wants to do them, so please find a way to make them consistent with the C++ API!!

Reply:
chrono and gloo are clearly independent piece of soft, with their own includes. So do you agree to keep packages for those ?
I can merge the "manual" function classe in main package, and I can also try to merge the cuda package. 
Would that be ok ?

Reply:
> chrono and gloo are clearly independent piece of soft, with their own includes. So do you agree to keep packages for those ?

Right, so use the names of those libraries, not the namespaces. That works fine for that purpose

> I can merge the packages for "manual" std::function instances in main package, and I can also try to merge the cuda package.

We already have separate torch and torch_cuda libraries, so that's the name we should give those packages, but like you said to keep backward compatibility, you left torch in org.bytedeco.pytorch and put torch_cuda in org.bytedeco.pytorch.cuda, so that's alright I guess. As for the std::function instances, yeah, I don't see why you want them in a separate package

Reply:
Ok, so moving `functins` only.
chrono is part of standard C++, and the name of the library is chrono. gloo lbrary name is gloo.
libtorch libraries are really torch_cpu and torch_cuda. torch is an empty library linking to both (or only one if pytorch was not compiled with cuda).

Reply:
> Ok, so moving `functins` only.

Ok, sounds consistent

> chrono is part of standard C++, and the name of the library is chrono. gloo lbrary name is gloo.
libtorch libraries are really torch_cpu and torch_cuda. torch is an empty library linking to both (or only one if pytorch was not compiled with cuda).

There is no "libchrono" though, it's libc++, libstdc++, etc...

Reply:
Could you try to move your chrono package here?
https://github.com/bytedeco/javacpp/tree/master/src/main/java/org/bytedeco/javacpp
If that works, it's probably a good place for that

Reply:
It needs to be completed to be anything useful, but I think it's a good thing to mutualize it somewhere.
Why not directly in JavaCPP. But since it'll concern few people and will grow if completed enough, isn't it better to put it in a separate presets ?

Reply:
The JavaCPP presets is meant to do C++ itself. Trying to chunk down of C++ is not something I want to do, no. I keep telling to come up with some sort of consistent way to using packages and you just keep trying to make it more complicated with no rationale. Try to start with the rationale

Reply:
So, let's start with how big you anticipate those "C++ presets" to become at some point in the future. Is it close to 1 MB, 10 MB, 100 MB, or 1 GB?